helper/schema/resource_config_data.go

```diff
@@ -0,0 +1,66 @@
+package schema
+
+import (
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// ResourceConfigGetter is a ResourceData-like interface for
+// ResourceValidateFunc implementors. It is different from ResourceData in that
+// it requires callers to check a second "known" return value before
+// interacting with any value.
+type ResourceConfigGetter interface {
+	// Get is the only function available to ResourceValidateFuncs.
+	//
+	// If known is true, value is guaranteed to be the type specified by the Schema
+	// for this field.
+	//
+	// If known is false, the value will always be nil, and the Config cannot be
+	// assumed to have a value one way or the other.
+	//
+	// It is the caller's responsibility to check known before using value, and
+	// to treat unknown values appropriately in validations.
+	Get(key string) (value interface{}, known bool)
```

Comment:
As we chatted about in Slack, I'd be worried about the important and different semantics of this `ok` with the `ResourceData` `ok`.

Namely, the latter means "non-zero" while the former means "you cannot assume anything about the value of this attribute".

Going to rework this to `GetIfKnown` w/ a nil check to see how that feels.